src/main/java/org/folio/circulation/domain/EffectiveCallNumberComponents.java

```diff
@@ -0,0 +1,41 @@
+package org.folio.circulation.domain;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+// Do not fail when an undefined property specified,
+// e.g. when inventory-storage has introduced a new property,
+// but circulation has not been updated
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class EffectiveCallNumberComponents {
```

Comment:
Personally I'd prefer we didn't have JSON serlization control attributes on domain classes.

I'm aware that the `toJson` and `fromJson` methods are a compromise here, at least they should be self contained.